SonarQube.Common/ProcessRunner.cs

```diff
@@ -64,6 +87,16 @@ public bool Execute(string exeName, string args, string workingDirectory, int ti
                 WorkingDirectory = workingDirectory
             };
 
+            if (envVariables != null)
+            {
+                foreach (KeyValuePair<string, string> envVariable in envVariables)
+                {
+                    if (!String.IsNullOrEmpty(envVariable.Value))
+                    {
+                        psi.EnvironmentVariables.Add(envVariable.Key, envVariable.Value);
```

Comment:
Tests: did you look at the existing ProcessRunnerTests? They create batch files on the fly that write output to the console that is checked by the test.